src/components/HeaderComp.vue

```diff
@@ -0,0 +1,35 @@
+<template>
+  <div class="nav">
+    <slot></slot>
+    <div class="options" @click="logout()">Logout</div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "HeaderComp",
+  methods: {
+    async logout() {
+      localStorage.clear();
+      this.$router.push({ name: "LoginPage" });
+      try {
+        await this.$store.dispatch("logout");
```

Comment:
put this line before redirecting to the loginpage.
try{
localStorage.clear();
await this.$store.dispatch("logout");
this.$router.push({ name: "LoginPage" });
}